Fix fuse_func method and fuse_custom_config_dict overwrite

sstsai-adl opened the pull request:
Summary: Fixing the interface so that the fuse_func and fuse_custom_config_dict is honored and not replaced but the default fuse_known_method and None.

Test Plan: Wait for sandcastle

Differential Revision: D40722395

jerryzh168 commented on torch/ao/quantization/fuse_modules.py:
```diff
@@ -160,8 +160,8 @@ def fuse_modules(model, modules_to_fuse, inplace=False, fuser_func=fuse_known_mo
-        fuser_func=fuse_known_modules,
-        fuse_custom_config_dict=None)
+        fuser_func=fuser_func,
+        fuse_custom_config_dict=fuse_custom_config_dict)
```
`fuse_custom_config_dict` is deprecated I think, we plan to add this to https://pytorch.org/docs/master/generated/torch.ao.quantization.backend_config.BackendConfig.html#torch.ao.quantization.backend_config.BackendConfig instead, and this part is not yet been designed

sstsai-adl replied:
Got it. Removed changes to fuse_custom_config_dict in an updated diff.

jerryzh168 commented:
actually, maybe add a test as well?

jerryzh168 commented:
can probably add it here: https://github.com/pytorch/pytorch/blob/master/test/quantization/eager/test_fuse_eager.py

sstsai-adl committed:
Fix fuse_func method overwrite (#87791)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/87791

Fixing the interface so that the fuse_func is honored and not replaced but the default fuse_known_method.

Test Plan: Wait for sandcastle

Reviewed By: jerryzh168

Differential Revision: D40722395

fbshipit-source-id: 3c2a0e12779095673e92cc6fc834eeb1e0a65ca8
